Userland/Libraries/LibVideo/VP9/BitStream.cpp

```diff
@@ -82,61 +108,85 @@ bool BitStream::read_bool(u8 probability)
         return_bool = true;
     }
 
-    while (m_bool_range < 128) {
-        bool new_bit;
-        if (m_bool_max_bits) {
-            new_bit = read_bit();
-            m_bool_max_bits--;
-        } else {
-            new_bit = false;
-        }
-        m_bool_range *= 2;
-        m_bool_value = (m_bool_value << 1u) + new_bit;
+    // Number of bits to read to shift the range above 128. Calculated by 7 - log2(n).
+    constexpr u8 range_shifts[127] = {
+        // 1..32
+        7, 6, 6, 5, 5, 5, 5, 4, 4, 4, 4, 4, 4, 4, 4, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 2,
+        // 33..64
+        2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 1,
+        // 65..96
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+        // 97..127
+        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1
+    };
```

Comment:
Is this some kind of optimization? If so, it could also be split into a seperate commit?

Reply:
Optional: A kinda neat trick for things like this to compute the values at compile time rather than hardcode the table. 
I believe something like this would work here (you may need to tweak it a little).
```c++
constexpr auto range_shifts = [] {
    Array<u8, 127> lut {};
    for (size_t n = 1; n <= lut.size(); n++)
        lut[n-1] = static_cast<u8>(7 - AK::log2(n));
    return lut;
}();
```

Reply:
I've cleaned up the constexpr initialization as you suggested, thanks!